lib/puppet/resource/catalog.rb

```diff
@@ -420,7 +429,8 @@ def to_data_hash
       'name'      => name,
       'version'   => version,
       'code_id'   => code_id,
-      'environment' => environment.to_s,
+      'catalog_uuid' => catalog_uuid,
+      'environment'  => environment.to_s,
```

Comment:
Similar, that serializing the catalog writes a hash containing the catalog_uuid?